### PR TITLE
feat: custom annotations

### PR DESCRIPTION
**Pull request by adityathebe**
resolves: https://github.com/flanksource/config-db/issues/639

### CODE REVIEW

File: scrapers/kubernetes/kubernetes.go [hunk 1]
```diff
@@ -144,16 +161,79 @@ func (kubernetes KubernetesScraper) Scrape(ctx api.ScrapeContext) v1.ScrapeResul
...
+	} else if item != nil && item.Config != nil {
+		expr, err := jp.ParseString("metadata.annotations")
+		if err != nil {
+			return "", "", err
+		}
+
+		data, err := oj.ParseString(*item.Config)
+		if err != nil {
+			return "", "", err
+		}
```
Comment (adityathebe) [inline scrapers/kubernetes/kubernetes.go:217]: @moshloop I think we should put all annotations as labels to avoid this parsing.

Comment (moshloop) [reply scrapers/kubernetes/kubernetes.go:217]: That would create too many labels and clutter the dropdowns

File: scrapers/kubernetes/kubernetes.go [hunk 2]
```diff
@@ -29,11 +31,23 @@ import (
...
+	AnnotationExcludeConfig = "config-db.flanksource.com/exclude"
+
+	// AnnotationExcludeChangeByType contains the list of change types to exclude
+	AnnotationExcludeChangeByType = "config-db.flanksource.com/exclude-changes"
+
+	// AnnotationExcludeChangeBySeverity contains the list of severity for the change types to exclude
+	AnnotationExcludeChangeBySeverity = "config-db.flanksource.com/exclude-change-severity"
+
+	// AnnotationCustomTags contains the list of tags to add to the scraped config
+	AnnotationCustomTags = "config-db.flanksource.com/tags"
```
Comment (moshloop) [inline scrapers/kubernetes/kubernetes.go:48]: These should be labelled ignore - (even if we use the term exclude in the code). 
We might want to rename our existing fields to ignore in the crd eventually - but grammatically exclude is still ok in that scenario

File: scrapers/kubernetes/kubernetes.go (hunk 1 above)
Comment (moshloop) [reply scrapers/kubernetes/kubernetes.go:217]: We can add the 2 change annotations as labels though - Searching on configs that have ignored changes still makes sense

**Pull request merged**